File: trainsim-client/src/models/Place.ts
export default class Place {
    private readonly _id: string;
    private readonly _stopId: string;
    private readonly _arriveAt: Date;
    private readonly _departAt: Date;

    constructor(id: string, stopId: string, arriveAt: Date, departAt: Date) {
        this._id = id;
        this._stopId = stopId;
        this._arriveAt = arriveAt;
        this._departAt = departAt;
    }

    public get id(): string {
        return this._id;
    }

    public get stopId(): string {
        return this._stopId;
    }

    public get arriveAt(): Date {
        return this._arriveAt;
    }
    
    public get departAt(): Date {
        return this._departAt;
    }
}

File: trainsim-itinerary/src/main/java/edu/drexel/trainsim/itinerary/models/Place.java
package edu.drexel.trainsim.itinerary.models;

import java.util.UUID;

public class Place {
    private final UUID id;
    private final String stopId;
    private final long arriveAt;
    private final long departAt;

    public Place(UUID id, String stopId, long arriveAt, long departAt) {
        this.id = id;
        this.stopId = stopId;
        this.arriveAt = arriveAt;
        this.departAt = departAt;
    }

    public UUID getId() {
        return id;
    }

    public String getStopId() {
        return stopId;
    }

    public long getArriveAt() {
        return arriveAt;
    }

    public long getDepartAt() {
        return departAt;
    }
}
